docs/interface.md

```diff
@@ -116,3 +116,34 @@ when a share is sent to a native application, the user agent may create an
 object with `ACTION_SEND`, setting the `EXTRA_SUBJECT` to `title`. Since Android
 intents do not have a URL field, `EXTRA_TEXT` would be set to `text` and `url`
 concatenated together.
+
+## Security considerations
+
+Implementations should observe the following security and privacy advice.
+
+Web Share enables data to be sent from websites to native applications. While
+this ability is not unique to Web Share, it does come with a number of potential
+security issues that may vary in severity (depending on the underlying
+platform).
+
+* Implementations MUST NOT allow the website to learn which apps are
+  installed, or which app was chosen from `navigator.share`. This information
+  could be used for fingerprinting, as well as leaking details about the user's
+  device.
+* On every call to `navigator.share`, the user MUST be presented with a dialog
+  asking them to select a target application (even if there is only one possible
+  target). This surface serves as a security confirmation, ensuring that
+  websites cannot silently send data to native applications.
+* Due to the capabilities of the API surface, websites MAY choose to restrict
+  `navigator.share` to secure browsing contexts (such as `https://` schemes).
+* Use of `navigator.share` from a [private browsing
+  mode](https://en.wikipedia.org/wiki/Privacy_mode) may leak private data to a
+  third-party application that does not respect the user's privacy setting.
+  Implementations should consider presenting additional warnings or disabling
+  the feature entirely when in a private browsing mode.
+* The data passed to `navigator.share` may be used to exploit buffer overflow
```

Comment:
Especially here: "may be used" indicates that it's a possibility. "can be used" means that it is able to be exploited, which is not necessarily true.